### PR TITLE
Convert `lib/timeout` to ES6

### PR DESCRIPTION
**Pull request by gu-stav**
## What does this change?

Converts `lib/timeout` to ES6.

## What is the value of this and can you measure success?

ES6!

## Does this affect other platforms - Amp, Apps, etc?

No.

## Screenshots

No.

## Tested in CODE?

No.

### CODE REVIEW

File: static/src/javascripts/lib/timeout.spec.js [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+    return new Promise(resolve => {
+        setTimeout(resolve, time);
+    });
+}
+
+describe('timeout', () => {
+    test(
+        "resolves with the promised value if the timeout doesn't fire",
+        done => {
+            timeout(10, Promise.resolve(12))
```
Comment (SiAdcock) [inline static/src/javascripts/lib/timeout.spec.js]: I wonder if we could use [`chance`](http://chancejs.com/) (installed in #16156) to take care of these magic numbers?

**Comment by gustavpursche** [comment]
@SiAdcock No more magic numbers!

File: static/src/javascripts/lib/timeout.spec.js [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
+                expect(error.message).toMatch(/timeout/i);
+                return wait(time);
+            })
+            .then(done)
+            .catch(done.fail);
+    });
+
+    test('rejects with timeout if promise resolves too late', done => {
+        const time = chance.integer({ min: 0, max: 10 });
+        const waitTime = chance.integer({ min: 0, max: 50 });
```
Comment (SiAdcock) [inline static/src/javascripts/lib/timeout.spec.js]: Is there a chance of `time < waitTime`? Maybe `waitTime` should take a minimum of `11`

Comment (gustavpursche) [inline static/src/javascripts/lib/timeout.spec.js]: You are right. Maybe this is the test-fail that we once saw, but never reproduced! Will change this now.

**Review by SiAdcock — APPROVED** [review]
👏

**Pull request merged**